File: backend/src/main/java/com/example/backend/entity/ShipmentTracking.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "shipment_tracking")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ShipmentTracking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String trackingNumber;
    private String shipmentStatus = "Preparing";
    private String carrier;

    private LocalDate estimatedDeliveryDate;
    private LocalDate deliveredDate;

    @OneToOne
    @JoinColumn(name = "order_id")
    private Order order;

    @Column(nullable = false)
    private Boolean isActive = true;
}

File: frontend/src/app/shared/models/shipment-tracking.ts
export interface ShipmentTracking {
  id: number;
  orderId: number;
  trackingNumber: string;
  shipmentStatus: 'PREPARING' | 'SHIPPED' | 'DELIVERED' | 'RETURNED';
  carrier: string;
  estimatedDeliveryDate?: Date;
  deliveredDate?: Date;
  isActive: boolean;
}
